dot/state/service.go

```diff
@@ -151,6 +153,21 @@ func (s *Service) Initialize(data *genesis.Data, header *types.Header, t *trie.T
 	return nil
 }
 
+func (s *Service) pruneStorage() {
+	for {
+		select {
+		case keys := <-s.Block.pruneKeyCh:
+			for _, v := range keys {
+				s.Storage.pruneKey(v)
+			}
+		case <-s.closeCh:
+			return
+		default:
+			time.Sleep(100 * time.Millisecond)
+		}
+	}
+}
```

Comment:
I think this function could go in `StorageState`, since `StorageState` contains the `blockState`